### PR TITLE
Fix for label_t to InstructionSeq mapping

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Fix for label_t to InstructionSeq mapping.

Resolve: #306 

Signed-off-by: Alan Jowett <alanjo@microsoft.com>

### CODE REVIEW

**Review by dthaler — APPROVED** [review]
Thanks, this should fix the issues I was running into

**Comment by Alan-Jowett** [comment]
Note:
The issue here is that unmarshalling the instructions from the raw BPF instruction sequence results in coalescing LDDW instructions (which span 2 PC locations) into a single LabeledInstruction. The result is that the index on an entry in a LabeledInstruction sequence doesn't match the pc of that instruction. This only shows up in programs with LDDW instructions.

File: src/crab_verifier.cpp [hunk 1]
```diff
@@ -113,13 +113,23 @@ static checks_db generate_report(cfg_t& cfg,
     return m_db;
 }
 
+auto get_line_info(const InstructionSeq& insts) {
+    std::map<int, std::optional<btf_line_info_t>> label_to_line_info;
+    for (auto& [label, inst, line_info] : insts) {
+        if (line_info.has_value())
+            label_to_line_info.insert({label.from, line_info});
```
Comment (elazarg) [inline src/crab_verifier.cpp:120]: Nit: I think `emplace` is better.

File: src/crab_verifier.cpp [hunk 2]
```diff
@@ -113,13 +113,23 @@ static checks_db generate_report(cfg_t& cfg,
...
+    auto label_to_line_info = get_line_info(prog);
     os << "\n";
     for (auto [label, messages] : db.m_db) {
         for (const auto& msg : messages) {
-            auto line_info = std::get<2>(prog[label.from]);
-            if (line_info.has_value()) {
-                auto& [file, source, line, _] = line_info.value();
+            auto line_info = label_to_line_info.find(label.from);
+            if (line_info != label_to_line_info.end()) {
+                auto& [file, source, line, _] = (*line_info).second.value();
```
Comment (elazarg) [inline src/crab_verifier.cpp:132]: Nit: Why not `*line_info->second`?

**Review by elazarg — APPROVED** [review]
LGTM. The two comments are out of curiosity.

**Pull request merged**